frontend/src/pages/analytics/segments/UserSegmentDetails.tsx

```diff
@@ -0,0 +1,616 @@
+                          cell: (e: SegmentJobStatusItem) =>
+                            new Date(e.jobStartTime).toLocaleString(),
+                        },
+                        {
+                          id: 'endTime',
+                          header: t('analytics:segment.details.endTime'),
+                          cell: (e: SegmentJobStatusItem) =>
+                            e.jobEndTime === 0
+                              ? '-'
+                              : new Date(e.jobEndTime).toLocaleString(),
+                        },
+                        {
+                          id: 'userNumber',
+                          header: t('analytics:segment.details.userNumber'),
+                          cell: (e: SegmentJobStatusItem) =>
+                            e.jobStatus === SegmentJobStatus.COMPLETED
+                              ? e.segmentUserNumber
+                              : '-',
+                        },
+                        {
+                          id: 'percentageOfTotalUser',
+                          header: t(
+                            'analytics:segment.details.percentageOfTotalUser'
+                          ),
+                          cell: (e: SegmentJobStatusItem) => {
+                            if (e.jobStatus !== SegmentJobStatus.COMPLETED) {
+                              return '-';
+                            }
+
+                            const percentage =
+                              (e.segmentUserNumber / e.totalUserNumber) * 100;
+                            return percentage.toFixed(2) + '%';
+                          },
+                        },
+                        {
+                          id: 'export',
+                          header: t('analytics:segment.details.export'),
+                          cell: (e: SegmentJobStatusItem) => {
+                            return (
+                              <Button
+                                variant="icon"
+                                iconName="download"
+                                disabled={
+                                  e.jobStatus !== SegmentJobStatus.COMPLETED
+                                }
+                                onClick={async () => {
+                                  const segmentApiResponse =
+                                    await getExportFileS3Url({
+                                      projectId,
+                                      appId,
+                                      segmentId,
+                                      jobRunId: e.jobRunId,
+                                    });
+
+                                  if (segmentApiResponse.success) {
+                                    const url =
+                                      segmentApiResponse.data.presignedUrl;
+                                    console.log(url);
+
+                                    const link = document.createElement('a');
```

Comment:
Should we can covert this to one utils function for download?

Reply:
done